src/FSharp.Data.GraphQL.Shared/TypeSystem.fs

```diff
@@ -1530,13 +1532,22 @@ and InputObjectDefinition<'Val> =
       Description : string option
       /// Function used to define field inputs. It must be lazy
       /// in order to support self-referencing types.
-      FieldsFn : unit -> InputFieldDef [] }
+      FieldsFn : unit -> InputFieldDef []
+      /// Optional function to customize default coercion behaviour
+      CoerceInput : (Value -> 'Val option) option }
```

Comment:
Have you considered using `Result<,>`?
Or @jberzy are there other places to move to `Result` so it is better to have a new PR for that?

Reply:
I would also prefer result, although the current design is based around options (see coerce input methods)